### ISSUE TITLE
How to correctly understand the meaning of ci output

### ISSUE THREAD

**Issue by marksugar** [wagoodman/dive#158]
```
Fetching image...
Parsing image...
  ├─ [layer:  1] 51e13bd688e9479 : [==============================>] 100 % (48/48)
  ├─ [layer:  2] fa85b118a7e6b06 : [==============================>] 100 % (477/477)
  ╧
Analyzing image...
  efficiency: 99.9397 %
  wastedBytes: 45967 bytes (46 kB)
  userWastedPercent: 1.0426 %
Run CI Validations...
  Using default CI config
  PASS: highestUserWastedPercent
  SKIP: highestWastedBytes: rule disabled
  PASS: lowestEfficiency
Result:PASS [Total:3] [Passed:2] [Failed:0] [Warn:0] [Skipped:1]
```
It seems to be wasting 45967 bytes, but it doesn't seem to be much.
But here is PASS: lowestEfficiency
I can't understand this output.

Can you help me analyze the meaning behind this write output?

**Comment by AubreySLavigne** [comment]
The CI tool is configured with the following thresholds:
* `lowestEfficiency: 90%`
* `highestWastedBytes: "disabled"`
* `highestUserWastedPerfect: 10%`

The sample run used by the CI "wastes" 46k, but this is within the acceptable range (99.94% > 90%). 

Source: [runtime/ci/evaluator.go:18](https://github.com/wagoodman/dive/blob/master/runtime/ci/evaluator.go#L18)

**Comment by wagoodman** [comment]
Thanks, @AubreySLavigne , that just about sums up the output, yup. I'll dig into a few more details regarding the metrics... I couldn't find a single metric that made sense in all cases, so I gave the 3 rules that I think someone would want to use:
  - **efficiency**: This is essentially `(total image size) / ( sum(bytes in all layers) )`. The idea is that if you don't remove any files or add any files twice, then you have an "efficiency" of 100%. If you start duplicating/deleting files, then that is counted against the efficiency (weighted by size of the wasted file).
  - **wastedBytes**: the raw number of bytes that were found to be either repeated on multiple layers, or found to be deleted in a higher layer (thus probably shouldn't be in any layer if it ultimately isn't used.
  - **userWastedPercent**: This is essentially the inverse of efficiency, except any modifications from the base image are not considered. More specifically, this is `wastedBytes / sum(bytes in all layers, except the base image layer)`.

This covers a few cases:
- If I have control over the base image, then I care that the overall `efficiency` remains high (above some threshold, `lowestEfficiency`).
- If I don't have control over the base image, then I care only about the layers I defined in my Dockerfile. This is, I want to make certain the `userWastedPercent` stays low (below some threshold, `* highestUserWastedPercent`).
- I don't care about any relative measure (like the other metrics), but I do know I want to keep the overall wasted bytes low. In this case, you just want to make certain you are minimizing the `wastedBytes` (stay below some threshold, `highestWastedBytes`). (There isn't a way to pick a universally-acceptable value for this metric for it to "work" in all cases, so I have this disabled by default.)

I'm up for changing the phrasing to make this output more understandable.
